leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -167,11 +169,25 @@ public Boolean call() throws Exception {
             Future<Boolean> pingFuture = pingCompletionService.poll(
                     leaderPingResponseWaitMs,
                     TimeUnit.MILLISECONDS);
-            return pingFuture != null && pingFuture.get();
+            if (pingFuture != null) {
+                if (pingFuture.get()) {
+                    leaderLog.trace("Successfully pinged " + leaderName);
+                        return true;
+                } else {
+                    leaderLog.info("Pinged " + leaderName + " and it reported that it is no longer the leader");
+                    return false;
+                }
+            } else {
```

Comment:
Optional: you could cut down on indentation by inverting this if, and then doing `} else if (pingFuture.get()) {`.